src/main/java/de/bsi/secvisogram/csaf_cms_backend/json/AdvisoryWrapper.java

```diff
@@ -444,68 +472,84 @@ public AdvisoryWrapper addEntryForNewCreatedVersion(String summary, String legac
-        if (getVersioningStrategy().getVersioningType() == VersioningType.Semantic && isInitialPublicReleaseOrEarlier()) {
-            ObjectNode entry = historyNode.addObject();
-            String now = DateTimeFormatter.ISO_INSTANT.format(Instant.now());
-            entry.put("date", now);
-            if (legacyVersion != null && !legacyVersion.isBlank()) {
-                entry.put("legacy_version", legacyVersion);
-            }
-            entry.put("number", this.getDocumentTrackingVersion());
-            entry.put("summary", summary);
-        } else {
-            ObjectNode latestEntry = getLatestEntryInHistoryAfterPrerelease(historyNode);
-            if (latestEntry == null) {
-                latestEntry = historyNode.addObject();
-            }
-            latestEntry.put("date", this.getDocumentTrackingCurrentReleaseDate());
-            if (legacyVersion != null && !legacyVersion.isBlank()) {
-                latestEntry.put("legacy_version", legacyVersion);
-            }
-            latestEntry.put("number", this.getDocumentTrackingVersion());
-            latestEntry.put("summary", summary);
+    public AdvisoryWrapper editLastRevisionHistoryEntry(String summary, String legacyVersion) {
+        return editLastRevisionHistoryEntry(summary, legacyVersion, null);
+    }
+
+    public AdvisoryWrapper editLastRevisionHistoryEntry(String summary, String legacyVersion, String releaseDate) {
+        ObjectNode lastHistoryNode = getLastHistoryNodeByDate();
+        if (releaseDate != null && !releaseDate.isBlank()) {
+            lastHistoryNode.put("date", this.getDocumentTrackingCurrentReleaseDate());
         }
+        if (legacyVersion != null && !legacyVersion.isBlank()) {
+            lastHistoryNode.put("legacy_version", legacyVersion);
+        }
+        lastHistoryNode.put("number", this.getDocumentTrackingVersion());
+        lastHistoryNode.put("summary", summary);
         return this;
     }
 
-    public void removeAllRevisionHistoryEntries() {
-
-        ArrayNode historyNode = getOrCreateHistoryNode();
-        historyNode.removeAll();
+    public String getLastRevisionHistoryEntrySummary() {
+        ObjectNode lastHistoryNode = getLastHistoryNodeByDate();
+        return lastHistoryNode.get("summary").asText();
     }
 
-    public void removeAllPrereleaseVersions() {
-
-        if (getVersioningStrategy().getVersioningType() == VersioningType.Semantic && isInitialPublicReleaseOrEarlier()) {
-            ArrayNode historyNode = getOrCreateHistoryNode();
-            historyNode.removeAll();
-        }
+    public AdvisoryWrapper setLastRevisionHistoryEntryNumber(String newNumber) {
+        ObjectNode lastHistoryNode = getLastHistoryNodeByDate();
+        lastHistoryNode.put("number", newNumber);
+        return this;
     }
 
-    private ObjectNode getLatestEntryInHistoryAfterPrerelease(ArrayNode historyNode) {
+    private ObjectNode getLastHistoryNodeByDate() {
```

Comment:
Here we call it "Node", maybe we can use a consistent wording. Be it "Node", "Element" or something else.

Reply:
I updated it to consistently use "element".